### DISCUSSION TITLE
Comparison to similar gems?

### DISCUSSION THREAD

**Discussion by redbar0n** [yippee-fun/phlex#86] [General]
Just wondering about this library’s raison d’etre? What did you find lacking in view_component gem so you decided to create phlex?

**Comment by joeldrapper** [comment]
Thanks for the question. We should definitely cover this in the docs. Phlex has a few big advantages over ViewComponent:

1. Phlex is a component abstraction that didn't depend on external ERB template files. Everything about a template can be encoded in pure Ruby and as a result, there's much less context switching.
2. Phlex is quite a bit faster than ViewComponent and we have plans to make significant improvements on this. Given that rendering views typically takes 50-80% of the response time for a Rails app, we can have a huge overall impact by working on this front.
3. Phlex supports component with DSLs, enabling you to expose a really rich API for your components. A good example of this is the table fabricator component in our README. This component takes an Enumerable resource and lets you define columns / rows with a header and a proc that's yielded items from the Enumerable.

```ruby
component Table, @articles, layout: :column do |t|
  t.data("Title") { a _1.title, href: article_path(_1) }
  t.data("Author", &:author)
  t.data("Published", &:published_at)
end
```
